<launch>

  <arg name="camera" default="/stereo_down/left"/>
  <arg name="bagfile" default="/home/uib/bagfiles/illetes/2020_01_16_illetes_tuberies/2020-01-16-12-20-37.bag"/>

  <param name="/use_sim_time" value="true"/>
  <node pkg="rosbag" type="play" name="rosbag" args="--clock $(arg bagfile)"/>

  <!-- Republishes the compressed topic so it can be used to image_proc node -->
  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=$(arg camera)/image raw out:=$(arg camera)/image_raw" />

  <!-- Run image proc -->
  <node ns="$(arg camera)" pkg="image_proc" type="image_proc" name="image_proc" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_left" args="0.4 0 0.86 0 1.57079632679 0 turbot/base_link turbot/stereo_down/left_optical 30"/>

  <!-- Viso2 -->
  <node pkg="viso2_ros" type="mono_odometer" name="mono_odometer" output="screen">
    <remap from="image" to="$(arg camera)/image_rect"/>
    <remap from="altitude" to="/turbot/navigator/altitude_filtered"/>
    <param name="base_link_frame_id" value="turbot/base_link"/>
    <param name="odom_frame_id" value="/odom"/>

    <param name="max_features" value="10" />
    <param name="motion_threshold" value="350" />
    <param name="camera_height" value="2.1" />
    <param name="camera_pitch"  value="0.00" />
  </node>

</launch>
